instrumentation/restlet/restlet-2.0/library/src/main/java/io/opentelemetry/instrumentation/restlet/v2_0/TracingFilter.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.restlet.v2_0;
+
+import static io.opentelemetry.instrumentation.api.servlet.ServerSpanNaming.Source.CONTROLLER;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.servlet.ServerSpanNaming;
+import io.opentelemetry.instrumentation.restlet.v2_0.internal.RestletServerSpanNaming;
+import org.restlet.Request;
+import org.restlet.Response;
+import org.restlet.routing.Filter;
+
+final class TracingFilter extends Filter {
+
+  private final Instrumenter<Request, Response> instrumenter;
+  private final String path;
+
+  public TracingFilter(Instrumenter<Request, Response> instrumenter, String path) {
+    this.instrumenter = instrumenter;
+    this.path = path;
+  }
+
+  @Override
+  public int doHandle(Request request, Response response) {
+
+    Context parentContext = Context.current();
+    Context context = parentContext;
+
+    Scope scope = null;
+
+    if (instrumenter.shouldStart(parentContext, request)) {
+      context = instrumenter.start(parentContext, request);
+      scope = context.makeCurrent();
+    }
+
+    ServerSpanNaming.updateServerSpanName(
+        context, CONTROLLER, RestletServerSpanNaming.SERVER_SPAN_NAME, path);
+
+    Throwable statusThrowable = null;
+    try {
+      super.doHandle(request, response);
+    } catch (Throwable t) {
+      statusThrowable = t;
+    }
+
+    if (scope == null) {
+      return CONTINUE;
+    }
+
+    scope.close();
+
+    if (response.getStatus() != null && response.getStatus().isError()) {
+      statusThrowable = response.getStatus().getThrowable();
+    }
+
+    instrumenter.end(context, request, response, statusThrowable);
```

Comment:
Shouldn't you rethrow the `statusThrowable` here? You're catching it in line 49, so wouldn't swallowing it here affect the instrumented app?

Reply:
I think it would be best to rethrow, but I can’t do it directly since `Filter`’s `doHandle` doesn’t declare `throws`. From what I’ve investigated, we have these options
* Catch `Throwable` and rethrow it wrapped as Restlet’s `ResourceException` (but throwing this in `Filter` could be confusing) or just `RuntimeException`
* Catch only `ResourceException` and then close scope & end span in a finally block - since Restlet’s handle does not declare throws and Resource throws only `ResourceException`
Each of those approaches has its drawbacks, but I think the second option is a bit better - in most cases the throwable is retrieved from the `Response` anyway

Reply:
If the caught exception is an `Error` or a `RuntimeException` you can rethrow it as is, otherwise you can wrap it with `ResourceException`